cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/support/HardTimeoutExpirationPolicy.java

```diff
@@ -63,7 +63,7 @@ public void init() {
     @Override
     public boolean isExpired(final TicketState ticketState) {
         return (ticketState == null) || ticketState.getCreationTime()
-          .plus(this.timeToKillInMilliSeconds, ChronoUnit.MILLIS).isAfter(ZonedDateTime.now(ZoneOffset.UTC));
+          .plus(this.timeToKillInMilliSeconds, ChronoUnit.MILLIS).isBefore(ZonedDateTime.now(ZoneOffset.UTC));
```

Comment:
Exactly, if creation time + max timetolive is before now, the ticket is expired.

```
creation date   +   ttl             now
|--------------------|               X            => expired

creation date      now     + ttl
|-------------------X---------|                   => still alive
```